cmd/juju/commands/bootstrap.go

```diff
@@ -598,7 +607,7 @@ func (c *bootstrapCommand) handleCommandLineErrorsAndInfoRequests(ctx *cmd.Conte
 	return false, nil
 }
 
-func (c *bootstrapCommand) cloud(ctx *cmd.Context) (*jujucloud.Cloud, error) {
+func (c *bootstrapCommand) cloud(ctx *cmd.Context) (string, *jujucloud.Cloud, error) {
```

Comment:
Do you mean that the name gets returned first? To me it reads like returning the name and the value - that's why I picked that ordering.

I thought about adding `Name` to `Cloud`. This makes the most sense to me, but it has implications for serialisation that I'm not happy about - should I also add `Name` to the internal `cloud` type for marshalling? I don't want to change the format of `clouds.yaml`. Adding Name.Cloud but then only populating it on the way through this function seems confusing too.

I toyed with just changing c.Cloud but decided it was nasty to mutate the bootstrapCommand as a side-effect of returning a Cloud, which is why I returned the name instead.